### PR TITLE
chore(button): migrate to vivid next

### PR DESCRIPTION
**Pull request by yinonov**
BREAKING CHANGE: removed connotations - announcement, info

BREAKING CHANGE: single affix icon support only

- scaffolding
- tests
- if no label but has icon should set class to apply icon-only styles
  - state applies class
  - add style according to class
- disabled style

### CODE REVIEW

File: src/components/button/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+### Outlined button with connotation
+
+```html preview
+<vwc-button appearance="outlined" label='primary' connotation='primary'></vwc-button>
+<vwc-button appearance="outlined" label='cta' connotation='cta'></vwc-button>
+<vwc-button appearance="outlined" label='success' connotation='success'></vwc-button>
+<vwc-button appearance="outlined" label='alert' connotation='alert'></vwc-button>
+<vwc-button appearance="outlined" label='announcement' connotation='announcement'></vwc-button>
+<vwc-button appearance="outlined" label='info' connotation='info'></vwc-button>
+```
```
Comment (rachelbt) [inline src/components/button/README.md:110]:
```suggestion
```## States
### Disabled
```html preview
<vwc-button label='ghost' appearance='ghost' disabled></vwc-button>
<vwc-button label='filled' appearance='filled' disabled></vwc-button>
<vwc-button label='outlined' appearance='outlined' disabled></vwc-button>
```
```

File: src/components/button/button.template.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+const getClasses = ({
+	connotation, appearance, shape, size, iconTrailing,
+}: Button) => classNames(
+	'control',
+	['icon-trailing', iconTrailing],
+	[`connotation-${connotation}`, Boolean(connotation)],
+	[`appearance-${appearance}`, Boolean(appearance)],
+	[`shape-${shape}`, Boolean(shape)],
+	[`size-${size}`, Boolean(size)],
```
Comment (rachelbt) [inline src/components/button/button.template.ts:15]: disabled class is needed and style as well :)

**Comment by yinonov** [comment]
@jshenkman should we support loading "state"?

**Comment by yinonov** [comment]
@jshenkman are `tonal` buttons in consideration? should they replace `ghost`? maybe `tonal` only apply in **toggle-buttons**?

**Comment by yinonov** [comment]
tests fail when running both _badge_ and _button_.
tests pass when removing `badge.spec.ts`.
@YonatanKra anything u can think of?

**Comment by yinonov** [comment]
@jshenkman FYI
meetings sign in button loading state

https://user-images.githubusercontent.com/15140652/149282886-2526ff80-1816-4f92-aa4d-df8f450e3d21.mov

File: src/components/button/button.spec.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	});
+
+	describe('size', () => {
+		it('sets correct internal size style', async () => {
+			const size = 'small';
+			(element as any).size = size;
+			await elementUpdated(element);
+
+			const control = element.shadowRoot.querySelector(`.control.size-${size}`);
+			expect(control).toBeInstanceOf(Element);
```
Comment (YonatanKra) [inline src/components/button/button.spec.ts:103]: What does this test mean?

File: src/components/button/button.spec.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+		});
+	});
+
+	describe('icon-only', () => {
+		it('sets correct internal icon-only style', async () => {
+			element.icon = 'home';
+			await elementUpdated(element);
+
+			const control = element.shadowRoot.querySelector('.control.icon-only');
```
Comment (YonatanKra) [inline src/components/button/button.spec.ts:112]: What's the other alternative? If I don't set an icon, what happens then?

File: src/components/button/button.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+ */
+type ButtonSize = Extract<Size, Size.BaseSmall | Size.Base | Size.BaseLarge>;
+
+/**
+ * Base class for button
+ *
+ * @public
+ */
+export class Button extends FoundationButton {
+	/**
```
Comment (YonatanKra) [inline src/components/button/button.ts:49]: As a side note - these comments in the code are driving me nuts!

**Review by YonatanKra — CHANGES_REQUESTED** [review]
See the comments

File: src/components/button/button.spec.ts (hunk 4 above)
Comment (yinonov) [reply src/components/button/button.spec.ts:112]: it won't set `icon-only` class on `control`

File: src/components/button/button.ts (hunk 5 above)
Comment (yinonov) [reply src/components/button/button.ts:49]: JSDoc / TSDoc?

Comment (YonatanKra) [reply src/components/button/button.ts:49]: I know - I don't like them :)
Not critical. Just leaving on the table :)

File: src/components/button/button.spec.ts (hunk 4 above)
Comment (YonatanKra) [reply src/components/button/button.spec.ts:112]: Might need to check before and after.

Comment (YonatanKra) [reply src/components/button/button.spec.ts:112]:
```
const hasIconOnlyBefore = checkIconOnly(element);
element.icon = 'home';
await elementUpdated(element);
const hasIconOnlyAfter = checkIconOnly(element);
expect(hasIconOnlyBefore).toEqual(false);
expect(hasIconOnlyAfter).toEqual(true);
```

File: src/components/button/button.spec.ts (hunk 3 above)
Comment (yinonov) [reply src/components/button/button.spec.ts:103]: don't we prefer repeat the same structure?
won't it be incoherent to have unique naming for each test?
it's the same pattern

File: src/components/button/button.ts (hunk 5 above)
Comment (yinonov) [reply src/components/button/button.ts:47]: "this is all over the town"

**Comment by yinonov** [comment]
icon figure element does not respond well to
```css
display: flex;
width: 1em;
height: 1em;
```
in safari, flex doesn’t set the flow to be blocked (in opposed to chrome / firefox)
it results in a “non-sized” element

if set to **block** instead of **flex**, the `<figure>` element is, again!, not vertically aligned to exact middle

image

**Review by YonatanKra — APPROVED** [review]

**Pull request merged**